packages/web/src/components/premium-content-purchase-modal/PremiumContentPurchaseModal.tsx

```diff
@@ -258,6 +267,18 @@ export const PremiumContentPurchaseModal = () => {
         ? PurchaseVendor.COINFLOW
         : PurchaseVendor.STRIPE
     })
+  const handleFormSubmit = useCallback(
```

Comment:
Wrapper required to ensure validation happens before guest creation while user creation action sagas are still in web/common